src/Microsoft.Health.ControlPlane.Core/Features/Rbac/RbacService.cs

```diff
@@ -21,18 +22,28 @@ public RbacService(IControlPlaneDataStore controlPlaneDataStore)
             _controlPlaneDataStore = controlPlaneDataStore;
         }
 
+        public async Task DeleteIdentityProviderAsync(string name, string eTag, CancellationToken cancellationToken)
+        {
+            await _controlPlaneDataStore.DeleteIdentityProviderAsync(name, eTag, cancellationToken);
```

Comment:
We should verify that you're not deleting the last identity provider when deleting. This may be easiest handled by a stored proc because we'll also need to clean up role memberships once they are implemented. Examples of stored procs are in the Fhir control plane.